File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第3章小明的衣服.cpp
//小明的衣服  lanqiaoOJ 1228 

#include<bits/stdc++.h>
using namespace std;
typedef long long ll;
int main(){
    ll n; cin>>n;
    priority_queue< ll,vector<ll>,greater<ll> > q;
    for(ll i=0;i<n;i++){
        ll a; cin>>a;
        q.push(a);
    }
    ll sum=0;
    while(q.size()>1){
        ll x = q.top(); q.pop();
        ll y = q.top(); q.pop();
        ll z = x+y;
        sum += z;
        q.push(z);
        cout <<z<<"  ";
    }
    cout<<sum<<endl;
}

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第3章小明的衣服.py
#小明的衣服  lanqiaoOJ 1228

import queue
q = queue.PriorityQueue()

n = int(input())
a = list(map(int, input().split()))

for i in range(len(a)):  q.put(a[i])

sum = 0
while q.qsize()>1:
    t = q.get() + q.get()
    sum += t
    q.put(t)

print(sum)